app/src/main/java/com/hover/stax/bounties/BountyEmailFragment.kt

```diff
@@ -0,0 +1,114 @@
+                viewModel.uploadBountyUser(emailInput!!.text)
+                emailInput!!.setState(getString(R.string.bounty_uploading_email), AbstractStatefulInput.INFO)
+            } else {
+                emailInput!!.setState(getString(R.string.bounty_email_error), AbstractStatefulInput.ERROR)
+            }
+        } else {
+            showOfflineDialog()
+        }
+    }
+
+    private fun showOfflineDialog() {
+        StaxDialog(requireActivity())
+                .setDialogTitle(R.string.internet_required)
+                .setDialogMessage(R.string.internet_required_bounty_desc)
+                .setPosButton(R.string.btn_ok, null)
+                .makeSticky()
+                .showIt()
+    }
+
+    private fun showEdgeCaseErrorDialog() {
+        StaxDialog(requireActivity())
+                .setDialogMessage(getString(R.string.edge_case_bounty_email_error))
+                .setPosButton(R.string.btn_ok, null)
+                .showIt()
+    }
+
+    private fun validates(): Boolean {
+        if (emailInput!!.text == null) return false
+        val email = emailInput!!.text.replace(" ", "")
+        return email.matches("(?:[A-Za-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|\"(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21\\x23-\\x5b\\x5d-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])*\")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21-\\x5a\\x53-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])+)])".toRegex())
+    }
+
+    fun observeEmailResult() {
+        viewModel.bountyEmailLiveData.observe(viewLifecycleOwner, { responseMap ->
+            val entry = responseMap.entries.iterator().next()
+            val responseCode = entry.key
+            val message = entry.value
+            if (responseCode in 200..299) saveAndContinue() else {
+                logErrorAndReportToFirebase(TAG, message!!, null)
+                if (isAdded && networkMonitor.isNetworkConnected) showEdgeCaseErrorDialog() else setEmailError()
+            }
+        })
+    }
+
+    private fun setEmailError() {
+        logAnalyticsEvent(getString(R.string.bounty_email_err, getString(R.string.bounty_api_internet_error)), requireContext())
+        emailInput!!.isEnabled = true
+        emailInput!!.setState(getString(R.string.bounty_api_internet_error), AbstractStatefulInput.ERROR)
+    }
+
+    private fun saveAndContinue() {
+        logAnalyticsEvent(getString(R.string.bounty_email_success), requireContext())
+        saveString(BountyActivity.EMAIL_KEY, emailInput!!.text, requireActivity())
+        NavHostFragment.findNavController(this).navigate(R.id.action_bountyEmailFragment_to_bountyListFragment)
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        if (dialog != null && dialog!!.isShowing) dialog!!.dismiss()
```

Comment:
The dialog is always null since as it's not assigned to anything.

Reply:
Thanks, this is now fixed in recent push